File: src/main/kotlin/chapter03/lec13/Lec13Main.java
package chapter03.lec13;

class Lec13Main {
}
class JavaHouse{
    private String address;
    private LivingRoom livingRoom;

    JavaHouse(String address, LivingRoom livingRoom) {
        this.address = address;
        this.livingRoom = new LivingRoom(10);
    }

    public LivingRoom getLivingRoom() {
        return livingRoom;
    }

    // 중첩 클래스에서 static가 없을 경우 밖에서 호출이 가능하므로  중첩클래스에는 static을 붙여줘야한다.
    public /*static*/ class LivingRoom{
        private double area;

        public LivingRoom(double area) {
            this.area = area;
        }
        public String getAddress(){
            return JavaHouse.this.address;
        }
    }
}

File: src/main/kotlin/chapter03/lec13/Lec13Main.kt
package chapter03.lec13

fun main(){
    //val javaHouse = House("se")
}

class House(
    private var address: String,
    private var livingRoom: LivingRoom = LivingRoom(10.0),
    private var kitchenRoom: KitchenRoom
){
    // 코틀린에서 권장되는 중첩 클래스
    class LivingRoom(
        private var area: Double
    )

    //코틀린에서 권장되지않는 중첩 클래스
    // 기본 적으로 바깥 클래스를 참조 하지 않으면 바깥클래스를 참조하고 싶으면 inner 키워드를 추가한다.
    inner class KitchenRoom(
        private var area: Double
    ){
        val address: String
            get() = this@House.address // 바깥 클래스를 참조하는 방법
    }
}